proto/supplyfinder.proto

```diff
@@ -7,28 +7,16 @@ option objc_class_prefix = "SLF";
 
 package supplyfinder;
 
+service Finder {
+  rpc CheckFood (Request) returns (stream ShopInfo) {}
+}
+
 service Vendor {
-  // Sends a greeting
-  rpc SayHello (HelloRequest) returns (HelloReply) {}
-  //
   rpc CheckInventory (FoodID) returns (InventoryInfo) {}
 }
 
 service Supplier {
-  // Sends a greeting
-  rpc SayHello (HelloRequest) returns (HelloReply) {}
-
-  rpc CheckVendor (FoodID) returns (VendorInfo) {}
-}
-
-// The request message containing the user's name.
-message HelloRequest {
-  string name = 1;
-}
-
-// The response message containing the greetings
-message HelloReply {
-  string message = 1;
+  rpc CheckVendor (FoodID) returns (stream VendorInfo) {}
```

Comment:
currently I am interleaving the request to supplier and vendor: Once I receive a VendorInfo, I query the corresponding vendor using the information. Then I use the streaming to get the next VendorInfo. 
Comparing to collecting a list of VendorInfo and querying all the vendors at the same time, I believe the streaming approach balances the load of supplier server and vendor servers. 
In addition, supplier server usually takes more time finding the qualifying vendors (in real world scenario), and streaming reduces the waiting time.

Reply:
Does it really balance load though, because aren't you still sending the same amount of requests to vendors? :) Perhaps it may add some wait time in between each call to a vendor, since you're waiting on the streaming results, but the result is a delayed response to the client (or increased latency). 

Stream responses each represent a separate network message, whereas a single response with a repeated field is a single message. While streaming has its use cases (for example, for a service that does some work in between each response), using repeated fields is more conventional for common use cases.

Reply:
Yeah I agree. Streaming not necessarily improves the performance on the client side, but `repeated` might do. I will change it after I deploy it on GKE because I want a working version running on it first. Thank you for your comments!